internal/stringset/stringset.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package stringset implements a very basic set collection of strings. Items
+// are unique, and basic set operations of addition, deletion, and existence
+// testing are supported. Additionally, supports exporting the elements as a
+// slice, optionally with the values sorted in ascending order.
+//
+// While the functionality of this package could easily be expanded, it would
+// likely soon be better to use a more full-featured, public set package.
+package stringset
```

Comment:
I feel like most of functionality you could get by making a couple of the functions in this package to work on map[string]bool and drop the new type. So my inclination would be delay from creating the abstraction.

```
    x := make(map[string]bool)

    # Add an element.
    x["a"] = true
    
    # Remove an element.
    delete(x, "a")
    
    # Is it in the list?
    if x["a"] { fmt.Println("it is!") }

    # The two functions I'm thinking of.
    v := GetKeys(x)       # --> gives me []string
    v := SortedKeys(x)    # --> gives me []string, sorted.
```

Sure using struct{} instead of bool can give you less memory usage, but for the stuff we are talking about here, I don't think it matters. If we are looking at providing our own non-map implementation, or make the Set thread-safe, that deserves a package, but that doesn't seem the case.

Reply:
Actually I like that he is using an empty struct, it shows that this is written in go rather than ported from say python. @cmarcelo you are perfectly correct that the memory usage for this case is trivial, which is part of the reason I didn't comment on your use of string->bool maps elsewhere, and it does make the `if x["a"] { fmt.Println("it is!") }` case much nicer to read than ` if _,ok := x["a"]; ok { fmt.Println("it is!") }`

However the *reason* for it being an empty struct (to save space) rather than the more obvious `bool` should be explained in the file if we are going to keep it in.

Reply:
@cmarcelo I'm fine with changing it to this implementation.

As for the bool vs empty struct... I chose the empty struct when I was abstracting it behind a set interface, because of the space advantage. But from a readability and maintainability standpoint, it's awful.  I would much rather _use_ the bool.

YMMV, but for me priorities go "serious space efficiency requirements" > "readability/maintainability" > "trivial space efficiency".  IMHO, "evidence something was written originally in Go rather than ported from Python" seems like a very odd thing to desire to me, and doesn't fall on my priority list at all.